code/game/objects/structures/mirror.dm

```diff
@@ -374,7 +374,11 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror/broken, 28)
 
 /obj/structure/mirror/magic/lesser/Initialize(mapload)
 	// Roundstart species don't have a flag, so it has to be set on Initialize.
-	selectable_races = get_selectable_species().Copy()
+	for(var/datum/species/species_type in subtypesof(/datum/species))
+		var/datum/species/species = species_type
+		if(species.check_roundstart_eligible())
+			selectable_races[initial(species_type.name)] = species_type
```

Comment:
this code doesn't work, and i think the whole way the framework is set up is kinda misleading because i can see why you thought it would work.

this shouldn't do anything because you can't run a proc on a typepath (which is pretty much text)- meaning that you would just have a magic mirror with all of the options regardless due to the `return ..()` call which would just autopopulate it with every kind of species available since the length of `selectable_races` will be zero. the default configuration might mislead you into thinking that it's working the way it's meant to, but it falls apart if the server operator futzes with that.

anyways the fix is really simple

```suggestion
	for(var/datum/species/species_type in GLOB.species_list)
		var/datum/species/species = new species_type
		if(species.check_roundstart_eligible())
			selectable_races[initial(species_type.name)] = species_type
		qdel(species)
```

you probably don't need that `qdel()` at the end there since we don't hold any references to it and BYOND should auto-Garbage Collect it but it's always better to be safe rather than sorry.

Reply:
i could also be wrong in my analysis of the situation but you should still just apply my suggestion- it should be the way to fix it